utilities/general.py

```diff
@@ -171,3 +180,106 @@ def create_isvc_label_selector_str(isvc: InferenceService, resource_type: str, r
 
     else:
         raise ValueError(f"Unknown deployment mode {deployment_mode}")
+
+
+def get_pod_images(pod: Pod) -> List[str]:
+    """Get all container images from a pod.
+
+    Args:
+        pod: The pod to get images from
+
+    Returns:
+        List of container image strings
+    """
+    return [container.image for container in pod.instance.spec.containers]
+
+
+def validate_image_format(image: str) -> Tuple[bool, str]:
+    """Validate image format according to requirements.
+
+    Args:
+        image: The image string to validate
+
+    Returns:
+        Tuple of (is_valid, error_message)
+    """
+    if not image.startswith(Resource.ApiGroup.IMAGE_REGISTRY):
+        return False, f"Image {image} is not from {Resource.ApiGroup.IMAGE_REGISTRY}"
+
+    if not re.search(SHA256_DIGEST_PATTERN, image):
+        return False, f"Image {image} does not use sha256 digest"
+
+    return True, ""
+
+
+@retry(wait_timeout=60, sleep=5)
+def get_pods_by_labels(
+    admin_client: DynamicClient,
+    namespace: str,
+    label_selector: str,
+    expected_num_pods: int,
+) -> list[Pod]:
+    """
+    Get pods by label selector in a namespace.
+
+    Args:
+        admin_client: The admin client to use for pod retrieval
+        namespace: The namespace to search in
+        label_selector: The label selector to filter pods
+        expected_num_pods: The expected number of pods to be found
+    Returns:
+        List of matching pods
+
+    Raises:
+        ResourceNotFoundError: If no pods are found
+    """
+    pods = list(
+        Pod.get(
+            dyn_client=admin_client,
+            namespace=namespace,
+            label_selector=label_selector,
+        )
+    )
+    if not pods:
+        raise ResourceNotFoundError(f"No pods found with label selector {label_selector} in namespace {namespace}")
```

Comment:
You would need to add these exceptions to the decorator exceptions_dict. Otherwise, they would fail immediately if the conditions are met. Just the same way timeout sampler would, if you raise exceptions.